vendor/github.com/google/cadvisor/fs/fs.go

```diff
@@ -533,6 +533,21 @@ func (self *RealFsInfo) GetDirFsDevice(dir string) (*DeviceInfo, error) {
 	}
 
 	mount, found := self.mounts[dir]
+	// try the parent dir if not found until we reach the root dir
+	// this is an issue on btrfs systems where the directory is not
+	// the subvolume
+	for !found {
```

Comment:
The comment says that this is to fix btrfs issue, but the code logic added here isn't capped for btrfs only. Is there any potential issue on other systems?

Reply:
@dchen1107 a check for `mount.Fstype == "btrfs"` couldn't be used here when the assignment of `mount` has failed. but in any case it should be independent and also apply to other filesystems. i can think of e.g. `/var/lib/kubelet` being just a directory on a partition `/var` on ext4, and the logic would apply there as well. it's a best effort lookup on any system where the passed `dir` is not found in the partition map